### DISCUSSION TITLE
Generated Connection and Edge types are not recognized in IntelliJ

### DISCUSSION THREAD

**Discussion by LajosPolya** [Netflix/dgs-framework#1283] [Q&A] [answered]
I'm implementation Relay Pagination but my type's `Connection` and `Edge` are not recognized in my schema. The application is able to start up fine but I get an error message in IntelliJ.

**build.gradle.kts**

```import com.netflix.graphql.dgs.codegen.gradle.GenerateJavaTask

plugins {
    id("com.netflix.dgs.codegen") version "5.1.17"
}

dependencies {
    implementation(platform("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:5.2.4"))
    implementation("com.netflix.graphql.dgs:graphql-dgs-webflux-starter")
    implementation("com.netflix.graphql.dgs:graphql-dgs-extended-scalars")
    implementation("com.netflix.graphql.dgs:graphql-dgs-pagination")
}

tasks.withType<GenerateJavaTask> {
    schemaPaths = mutableListOf("$projectDir/src/main/resources/schema")
    packageName = "com.exaple.schema"
    generateClient = false
    typeMapping = mutableMapOf(
        Pair("MyCustomTypeConnection", "graphql.relay.SimpleListConnection<MyCustomType>")
    )
}
```

**schema.graphql**
I get an error in this file stating that `Unknown directive "connection"`.
I can remove this error by adding `directive @connection on OBJECT` as described here https://github.com/Netflix/dgs-framework/discussions/507

I also get the error `Unknown type "MyCustomTypeConnection"`
```
type MyCustomType @connection {
    id: Long
    name: String!
}

type Query {
  getMyCustomTypes(first: Int!, after: String): MyCustomTypeConnection
}
```

The code compiles and run fine. I'm able to fetch paginated items but I'd like to get rid of the error `Unknown type "MyCustomTypeConnection"`  in IntelliJ?

I also can't find the generated classes for the `Connection` and `Edge` types in the `build` directory.

**Comment by srinivasankavitha** [comment, marked as answer]
The `@connection` directive is specific to the DGS framework and treats it specially by internally generating the SDL type MyCustomTypeConnection at runtime. This is not generated at build time. Also note that this will not generate corresponding POJOs (and hence will not be present in the build directory). The expectation is for the user to map it to the corresponding relay  custom type. IntelliJ is unable to understand the runtime generated SDL type and that is why you see the error. 
To fix this , we will need to add logic in the DGS IntelliJ Plugin to detect this and suppress it. We have an issue to track that : https://github.com/Netflix/dgs-intellij-plugin/issues/30
